### DISCUSSION TITLE
PATH environment variable

### DISCUSSION THREAD

**Discussion by 0free** [microsoft/WSL#5099] [General] [answered]
Microsoft Windows [Version 10.0.18363.778]

This was the command:
`find /home/ubuntu/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/root-ipq806x/ -mindepth 1 -execdir touch -hcd "@1587126718" "{}" +
`

The was the error:
`find: The relative path 'Files/WindowsApps/CanonicalGroupLimited.UbuntuonWindows_1804.2020.5.0_x64__79rhkp1fndgsc' is included in the PATH environment variable, which is insecure in combination with the -execdir action of find.  Please remove that entry from $PATH`

How to solve this?

**Comment by dirk-seynhaeve** [comment]
What if you make the touch command "secure" by providing the full path for it?
```
find <location> -mindepth 1 -execdir /usr/bin/touch ....
```
That way, a relative directory in your PATH won't accidentally pick it up.

**Comment by therealkenc** [comment, marked as answer]
Looks like the relative path here is "`Files/WindowsApps/CanonicalGroupLimited....`" when "`/mnt/c/Program Files/WindowsApps....`" is intended. Somehow `$PATH` got borked by something before `find` was ever run. That something is almost always in (or called from) a customized `.bashrc`.

Possible remedies are:

1. Fix the script that's mangling the `$PATH` containing spaces.
2. Make the last thing you do in `.bashrc` setting `$PATH` explicitly (with or without Windows path components).
3. Put `appendWindowsPath=false` in [`/etc/wsl.conf`](https://devblogs.microsoft.com/commandline/automatically-configuring-wsl/) so there aren’t spaces to mangle.

The (3) option will mean you'll lose your windows path components, but you can fix that up with (2). Or do (1).

**Comment by ghost** [comment]
Non modified .bashrc here, temporarily removing windows path before make e.g:"export PATH="<commonlinuxpath>:!removewindowspaths!" fixed it. (substitute with the result of echo $PATH)

**Reply by thagabe** [reply]
This worked for me as a quick temp fix!

**Comment by perrog** [comment]
Is it a bug in the commands (e.g. find), and not really WSL? Or just a bad practice to have spaces in paths for PATH env variable?
I got similar strange errors also when running sudo command from my WSL with custom Linux distrubution (CentOS7).

```
$ sudo yum remove autoconf.noarch
env: Files: No such file or directory
$ sudo cat /etc/resolv.conf
env: Files: No such file or directory
```
